sdk/src/deserialize_utils.rs

```diff
@@ -0,0 +1,91 @@
+use serde::{Deserialize, Deserializer};
+
+pub fn default_on_eof<'de, T, D>(d: D) -> Result<T, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de> + Default,
+{
+    let result = T::deserialize(d);
+    match result {
+        Err(err) if err.to_string() == "io error: unexpected end of file" => Ok(T::default()),
+        result => result,
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use bincode::deserialize;
+
+    #[test]
+    fn test_default_on_eof() {
+        #[derive(Serialize, Deserialize, Debug, PartialEq)]
+        struct Foo {
+            bar: u16,
+            #[serde(deserialize_with = "default_on_eof")]
+            baz: Option<u16>,
+            #[serde(deserialize_with = "default_on_eof")]
```

Comment:
I'm curious what would happen if this second `default_on_eof` was omitted?   Does `deserialize` blow up?

Reply:
If the second `default_on_eof` is omitted: if `bar` and `baz` are both present, `deserialize` works fine, but if `deserialize` on a partial struct blows up.

Reply:
I can add a should-panic unit test to emphasize usage, if that would be helpful...?

Reply:
Yeah a test demoing the blow-up that we can point people at when their code blows up seems helpful.  Sucks that we can't have serde attributes depend on each other